cardano_node_tests/tests/tests_conway/test_drep.py

```diff
@@ -527,6 +575,315 @@ def test_no_witness_register_and_retire(  # noqa: C901
+                src_address=payment_addr.address,
+                use_build_cmd=True,
+                tx_files=tx_files,
+                deposit=deposit_amt,
+            )
+
+        err_msg = str(excinfo.value)
+        assert "StakeKeyNotRegisteredDELEG" in err_msg, err_msg
+        reqc.cli088.success()
+
+    @allure.link(helpers.get_vcs_link())
+    @submit_utils.PARAM_SUBMIT_METHOD
+    @common.PARAM_USE_BUILD_CMD
+    @pytest.mark.testnets
+    @pytest.mark.smoke
+    def test_drep_no_retirement_before_register(
+        self,
+        cluster: clusterlib.ClusterLib,
+        payment_addr: clusterlib.AddressRecord,
+        use_build_cmd: bool,
+        submit_method: str,
+    ):
+        """Test No Drep retirement before register.
+
+        * Create a retirement certificate without registering
+        * Submit certificate
+        * check it is not possible to retire before register
+        """
+        temp_template = common.get_test_id(cluster)
+        drep_keys = cluster.g_conway_governance.drep.gen_key_pair(
+            key_name=temp_template, destination_dir="."
+        )
+        drep_id = cluster.g_conway_governance.drep.get_id(
+            drep_vkey_file=drep_keys.vkey_file,
+            out_format="hex",
+        )
+        deposit = cluster.conway_genesis["dRepDeposit"]
+
+        reqc.cli089.start(url=helpers.get_vcs_link())
+        ret_cert = cluster.g_conway_governance.drep.gen_retirement_cert(
+            cert_name=temp_template,
+            deposit_amt=deposit,
+            drep_vkey_file=drep_keys.vkey_file,
+        )
+        tx_files_ret = clusterlib.TxFiles(
+            certificate_files=[ret_cert],
+            signing_key_files=[payment_addr.skey_file, drep_keys.skey_file],
+        )
+
+        # Expecting error for both cases as drep is not registered
+        if submit_method == "api":
+            with pytest.raises(submit_api.SubmitApiError) as excinfo_submit:
+                clusterlib_utils.build_and_submit_tx(
+                    cluster_obj=cluster,
+                    name_template=f"{temp_template}_ret",
+                    src_address=payment_addr.address,
+                    submit_method=submit_method,
+                    use_build_cmd=use_build_cmd,
+                    tx_files=tx_files_ret,
+                    deposit=-deposit,
+                )
+            err_msg = str(excinfo_submit.value)
+            err_msg = err_msg.replace("\\", "")
+        else:
+            with pytest.raises(clusterlib.CLIError) as excinfo_cli:
+                clusterlib_utils.build_and_submit_tx(
+                    cluster_obj=cluster,
+                    name_template=f"{temp_template}_ret",
+                    src_address=payment_addr.address,
+                    submit_method=submit_method,
+                    use_build_cmd=use_build_cmd,
+                    tx_files=tx_files_ret,
+                    deposit=-deposit,
+                )
+            err_msg = str(excinfo_cli.value)
```

Comment:
This can be simplified to handle both `SubmitApiError` and `CLIError` in single `pytest.raises` context.

Reply:
Initially, submit api and cli were giving two different error strings and mypy was complaining about types. 

Now It has been updated to handle both errors in single pytest.raises with common error message between both.